### ISSUE TITLE
Orbit Api Bug results in Duplicate data and bad reports

### ISSUE THREAD

**Issue by samhowes** [samhowes/foothills-data#34]
## Problem

Reports in the app are now "messed up":
![image](https://user-images.githubusercontent.com/6216665/137116570-fa3f2e2a-b832-4c10-8c94-3763dc389222.png)

## Analysis

[The last sync](https://github.com/samhowes/foothills-data/commit/60bd1c9b53ccd4df933b04114c8c6e9ca6416607/checks) appears to have uploaded more data than expected: 
```
Person:     Pushed: 4 in 2.20 seconds
CheckIn:    Pushed: 767 in 130.90 seconds
Donation:   Pushed: 24182 in 4037.50 seconds
Attendance: Pushed: 15645; Failed: 1 in 2705.30 seconds
Membership: Pushed: 4409 in 712.40 seconds
Note:       Pushed: 4012 in 674.10 seconds
```

After looking into some of the data, it looks like the api allowed some duplicate data to be uploaded:
![image](https://user-images.githubusercontent.com/6216665/137116024-eec950c9-ba2d-46c1-980c-728bd464ab08.png)

According to [the docs for activities](https://docs.orbit.love/reference/activity-overview), the `key` field is:
> A unique identitier for the activity that makes sure duplicates of it are not recorded. Optional but recommended if your integration may resend data multiple times. A strong key choice might be the id or timestamp of an event registration along with the event name, e.g. "july-conference-registration:123456". If Orbit receives a POST to create an activity with that key more than once for the same member, it will only create one.

However, from the above screenshot, there are clearly two activities with separate ids and create dates that have the same `key` field.

### New Activity types
For a while, I have noticed the [list of ActivityTypes](https://docs.orbit.love/reference/activity-type-object) coming back from the api have been empty. Now, however, it is returning some data. 

The activity on the left in the above screenshot maps to this activity type:
```json
    {
      "id": "WVsEKL",
      "type": "activity_type",
      "attributes": {
        "name": "Event Attendance",
        "short_name": "Led test event",
        "key": "event_attendance",
        "channel": "BelongingAndCommunity",
        "weight": "1.0"
      }
    },
```

The activity on the right, with activity_type id `orGs25` does not appear in the list of activity types for the workspace.

## Root Cause
I suspect the root cause of this issue to be related to Orbit's back end Activity Type processing. You can see from the screenshot diff above that the `id` field of the `activity_type` relationship is different for both the activities. 

In the report UI, it appears it is showing two different activity types as one: 
![image](https://user-images.githubusercontent.com/6216665/137116949-6bcf9818-e173-476f-a9d8-79adeeb222d7.png)

This could also be related to the recent cleaning of data that I did: I issued DELETEs for many of these activities, and uploaded activities in their place. If the api has a bug in its delete logic, it is reasonable to expect it to manifest in this way.

## Mitigation

Since this appears to be a bug that is counter to the Orbit api's documentation, the only solution for this appears to be implementing a workaround. I have pinged the Orbit discord developer channel, however I do not expect great results from that communication channel.

### Workaround
Until now, we have been relying on the Orbit api to reject a duplicate key with an error response to the HTTP request. If we can't rely on this response, we can change our uploading strategy to check if some data exists before uploading it. 

## Moving forward
There's no telling how much data is currently duplicated in the current workspace. Ensuring clean data will likely mean creating two fresh workspaces (one each for Adults and Children), and initializing those workspaces with fresh data, or issuing delete statements for all the existing activities in the current workspace. 

Given the mismatched ActivityTypes mentioned in [analysis](#new-activity-types) above, it is possible that there is something wrong with how we are uploading activity types. The api does not provide an explicit way to manage these types though, it appears to auto-create these objects based on the data that we have uploaded. So this issue may come up again in another way.

**Comment by samhowes** [comment]
@revsean please review the issue above. I suspect that all the activity types are impacted by this, and that until we address this, the current reports are not trustworthy or meaningful.

**Comment by revsean** [comment]
One thing to note is the duplicates are not identical... The activity type is different
![Screen Shot 2021-10-14 at 12 18 57 PM](https://user-images.githubusercontent.com/86415336/137373778-77d90410-e038-4117-a093-62bf198c1bb2.png)

**Comment by samhowes** [comment]
From the discord #developers channel:
> Chance from Orbit — Today at 2:47 PM
Hey @samhowes . Wanted to follow up. We have identified the issue and are working on a fix. We rolled out changes to how activity types are populated, resulting in a few duplicate activities during the migration window. We're planning to clean up the dupes in our back office and will reach out when complete

Looks like they plan to take care of this.
